lib/matplotlib/collections.py

```diff
@@ -899,6 +900,118 @@ def set_paths(self, paths):
     def get_paths(self):
         return self._paths
 
+    def legend_items(self, mode="colors", useall="auto", num=10,
```

Comment:
How far up the class hierarchy does it make sense to move this?

Reply:
This particular method only makes sense for the `PathCollection`. A less useful version of it could be created for any `Collection` and yes, even `Artist` could in principle have such method. However the point being that at least for more complex things like scatter (PathCollection), LineCollection, contour (ContourSet), imshow (AxesImage) there need to be some options for the user to decide on the final appearance. I'm unsure how to generalize such arguments and how to use them, other than letting the user call this method and supply the returned handles/labels to `legend` - see comment below.